stglib/core/utils.py

```diff
@@ -744,6 +745,79 @@ def loadmat(filename):
     data = spio.loadmat(filename, struct_as_record=False, squeeze_me=True)
     return _check_keys(data)
 
+def read_samplingrates_continuous(ds,conn):
+    '''
+    Reads in sample information from RBR instrument in continuous mode
+    '''
+    try:
+        samplingperiod = conn.execute(
+            "select samplingperiod from schedules").fetchall()[0][0]
+    except sqlite3.OperationalError:
+        samplingperiod = conn.execute(
+            "select samplingperiod from continuous").fetchall()[0][0]
+
+    samplingrate = round(1000/samplingperiod)   #convert from ms to sec
+    samplingperiod = 1./samplingrate     #period from rate (Hz)
```

Comment:
Why are you redefining `samplingperiod` here? I would suggest you get `samplingperiod` from the SQL query, then adjust its units to seconds, and then define the rate by the inverse of that value. Here you are getting at it in a roundabout way.

Reply:
Simplified, thanks